routers/web/user/setting/profile.go

```diff
@@ -357,6 +358,22 @@ func Appearance(ctx *context.Context) {
 	ctx.Data["Title"] = ctx.Tr("settings")
 	ctx.Data["PageIsSettingsAppearance"] = true
 
+	hiddenEvents := &forms.UpdateCommentTypeForm{}
+	eventsRaw, err := user_model.GetSettings(ctx.User.ID, []string{"hidden_comment_types"})
+	if err != nil {
+		ctx.ServerError("GetSettings", err)
+		return
+	}
+	if eventsRaw["hidden_comment_types"] != nil && eventsRaw["hidden_comment_types"].SettingValue != "" {
+		err = json.Unmarshal([]byte(eventsRaw["hidden_comment_types"].SettingValue), &hiddenEvents)
+		if err != nil {
+			ctx.ServerError("json.Unmarshal", err)
```

Comment:
If there is something wrong in database, there is no chance to set the value to the correct one.

Reply:
Yes, but what sould it do instead? Just use default values?

Reply:
I think default value is enough, just as there is no setting.